src/apispec/ext/marshmallow/openapi.py

```diff
@@ -147,105 +141,38 @@ def schema2parameters(
-        """Return an array of OpenAPI parameters given a mapping between field names and
-        :class:`Field <marshmallow.Field>` objects. If `default_in` is "body", then return an array
-        of a single parameter; else return an array of a parameter for each included field in
-        the :class:`Schema <marshmallow.Schema>`.
-
-        In OpenAPI3, only "query", "header", "path" or "cookie" are allowed for the location
-        of parameters. In OpenAPI 3, "requestBody" is used when fields are in the body.
-
-        This function always returns a list, with a parameter
-        for each included field in the :class:`Schema <marshmallow.Schema>`.
-
-        https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#parameterObject
-        """
-        parameters = []
-        body_param = None
-        for field_name, field_obj in fields.items():
-            if field_obj.dump_only:
-                continue
-            param = self.field2parameter(
+        return [
+            self._field2parameter(
                 field_obj,
                 name=self._observed_name(field_obj, field_name),
-                default_in=default_in,
+                location=location,
             )
-            if (
-                self.openapi_version.major < 3
-                and param["in"] == "body"
-                and body_param is not None
-            ):
-                body_param["schema"]["properties"].update(param["schema"]["properties"])
-                required_fields = param["schema"].get("required", [])
-                if required_fields:
-                    body_param["schema"].setdefault("required", []).extend(
-                        required_fields
-                    )
-            else:
-                if self.openapi_version.major < 3 and param["in"] == "body":
-                    body_param = param
-                parameters.append(param)
-        return parameters
+            for field_name, field_obj in fields.items()
+        ]
 
-    def field2parameter(self, field, *, name, default_in):
+    def _field2parameter(self, field, *, name, location):
         """Return an OpenAPI parameter as a `dict`, given a marshmallow
         :class:`Field <marshmallow.Field>`.
 
         https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#parameterObject
         """
-        location = field.metadata.get("location", None)
-        prop = self.field2property(field)
-        return self.property2parameter(
-            prop,
-            name=name,
-            required=field.required,
-            multiple=isinstance(field, marshmallow.fields.List),
-            location=location,
-            default_in=default_in,
-        )
-
-    def property2parameter(
-        self, prop, *, name, required, multiple, location, default_in
-    ):
-        """Return the Parameter Object definition for a JSON Schema property.
+        openapi_location = __location_map__.get(location, location)
```

Comment:
Even if we keep the mapping we shouldn't call it twice. The only reason that this had to be here was if the location was extracted from the field metadata where webargs used different strings.

Reply:
Right. Fixed.